From f2a658ab7d0bd1a9cf430c75e4a3998583c151e8 Mon Sep 17 00:00:00 2001
From: Baokun Li <libaokun1@huawei.com>
Date: Mon, 21 Nov 2022 12:02:15 +0800
Subject: [PATCH 0086/1398] [Huawei] Revert "[Huawei] ext4: fix bug_on in
 ext4_ext_determine_hole when eh_entries is zero"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187444

--------------------------------

This reverts commit bb550ca0eec95d425b66bf7793ffb9965c8e4bc6.
Roll back the patch and integrate it into the community solution
958b0ee23f5a ("ext4: fix bug in extents parsing when eh_entries == 0 and
eh_depth > 0").

Signed-off-by: Baokun Li <libaokun1@huawei.com>
---
 fs/ext4/extents.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/fs/ext4/extents.c b/fs/ext4/extents.c
index 88567836d5cc..1674b8522843 100644
--- a/fs/ext4/extents.c
+++ b/fs/ext4/extents.c
@@ -459,10 +459,6 @@ static int __ext4_ext_check(const char *function, unsigned int line,
 		error_msg = "invalid eh_entries";
 		goto corrupted;
 	}
-	if (unlikely((eh->eh_entries == 0) && (depth > 0))) {
-		error_msg = "contradictory eh_entries and eh_depth";
-		goto corrupted;
-	}
 	if (!ext4_valid_extent_entries(inode, eh, lblk, &pblk, depth)) {
 		error_msg = "invalid extent entries";
 		goto corrupted;
-- 
2.17.1

